upload-api/functions/ucan-invocation-router.js

```diff
@@ -108,90 +108,41 @@ async function ucanInvocationRouter(request) {
     taskBucket,
     workflowBucket,
     streamName,
-    kinesisClient
+    kinesisClient,
   }
 
-  // Decode body and its invocations
-  const body = Buffer.from(request.body, 'base64')
-  const invocations = await CAR.decode({
-    body,
-    // @ts-expect-error - type is Record<string, string|string[]|undefined>
-    headers: request.headers,
-  })
-
   // Process workflow
   // We block until we can log the UCAN invocation if this fails we return a 500
   // to the client. That is because in the future we expect that invocations will
   // be written to a queue first and then processed asynchronously, so if we
   // fail to queue the invocation we should not handle it.
-  await processWorkflow(body, processingCtx)
-
-  // Execute invocations
-  const results = await Promise.all(
-    invocations.map((invocation) => execute(invocation, server, {
-      signer: serviceSigner
-    }))
+  const incoming = await processAgentMessageArchive(
+    {
+      headers: /** @type {Record<string, string>} */ (request.headers),
+      body: Buffer.from(request.body, 'base64'),
+    },
+    processingCtx
   )
 
-  const forks = []
-  const out = []
-
-  for (const receipt of results) {
-    out.push(receipt.data.out.error || receipt.data.out.ok)
-    forks.push(processInvocationReceipt(receipt.bytes, processingCtx))
-  }
+  // Execute invocations
+  const outgoing = await Server.execute(incoming, server)
+  const response = CAR.response.encode(outgoing)
 
-  await Promise.all(forks)
-  const response = await CBOR.encode(out)
+  processAgentMessageArchive(response, processingCtx)
```

Comment:
we were awaiting on this before. Should we keep until we wire up aa dead letter queue?

Reply:
Ah good catch, I didn't intend to change this

```suggestion
  await processAgentMessageArchive(response, processingCtx)
```

Reply:
@Gozala this still did not get changed. Maybe commit got lost